src/reanimated2/shareables.ts

```diff
@@ -44,7 +44,7 @@ export function registerShareableMapping(
   _shareableCache.set(shareable, shareableRef || _shareableFlag);
 }
 
-function isPlainJSObject(object: object) {
+export function isPlainJSObject(object: unknown): object is object {
```

Comment:
Calling `Object.getPrototypeOf(null)` or `Object.getPrototypeOf(undefined)` throws a `TypeError` so it cannot be `unknown` I guess.

Reply:
Yeah, you are right, do you prefer to keep this only for `object` type and do the checks for `undefined` and `null` in `extractInputs` or make this function also accept `unknown` and do the checks here?

Reply:
IMO given that `value` in `shareables` that is passed to this function is `any` and `inputs` in `mappers` are `unknown`, it's better to do the checks in `isPlainJSObject`

Reply:
Let's leave it as it is now (`object`) and also make sure that you cannot pass `null` to it

Reply:
[0f7a4e8](https://github.com/software-mansion/react-native-reanimated/pull/5363/commits/0f7a4e85de00f9a4b0cccbf848d0f89f2bc7bfca)